python/python/tests/test_balanced.py

```diff
@@ -58,6 +58,32 @@ def balanced_dataset(tmp_path, big_val):
     )
 
 
+def test_write_fragments(balanced_dataset, tmp_path):
+    tbl = balanced_dataset._take_rows(range(10))
+    transaction = lance.fragment.write_fragments(
+        tbl,
+        tmp_path / "ds",
+        enable_move_stable_row_ids=True,
+        return_transaction=True,
+    )
+    operation = lance.LanceOperation.Overwrite(
+        transaction.operation.new_schema, transaction.operation.fragments
+    )
+    blob_operation = lance.LanceOperation.Overwrite(
+        transaction.blobs_op.new_schema, transaction.blobs_op.fragments
+    )
```

Comment:
Why do we get an overwrite transaction here?  Isn't the default `mode` set to `append`?

Note: I'm not sure if this is related to the PR or just how things worked previously but it was masked by the fact we were ignoring the txn op.  We can figure it out in a future PR.